keras/preprocessing/image.py

```diff
@@ -473,6 +474,10 @@ def __init__(self,
             self.channel_axis = 3
             self.row_axis = 1
             self.col_axis = 2
+        if validation_split is not None and not 0 <= validation_split <= 1:
```

Comment:
`not 0 < validation_split < 1` (0 and 1 are invalid values)

Reply:
@fchollet wouldn't it fail if `validation_split` defaults to 0.0, as suggested above?